arch/common/gen_isr_tables.py

```diff
@@ -181,7 +227,49 @@ def main():
                 error("Regular Interrupt %d declared with parameter 0x%x "
                         "but no SW ISR_TABLE in use"
                         % (irq, param))
-            swt[irq - offset] = (param, func)
+
+            if not "CONFIG_MULTI_LEVEL_INTERRUPTS" in syms:
+                swt[irq - offset] = (param, func)
+            else:
+                # In the code below, there are a few hardware independent
+                # magic numbers which represent various levels of interrrupts.
+                # 0x000000FF - represents the 1st level (i.e. the interrupts
+                #              that directly go to the processor).
+                # 0x0000FF00 - represents the 2nd level (i.e. the interrupts funnel
+                #              into 1 line which then goes into the 1st level)
+                # 0x00FF0000 - represents the 3rd level (i.e. the interrupts funnel
+                #              into 1 line which then goes into the 2nd level)
+
+                # Figure out third level interrupt position
+                debug('IRQ = ' + hex(irq))
+                irq3 = (irq & 0x00FF0000) >> 16
```

Comment:
I'd prefer if you defined these constants as globals at the top of the file.
Similar to how in C we #define them and then use the macros in code.
https://www.python.org/dev/peps/pep-0008/#constants

Reply:
ok